backends/dnf/pk-backend-dnf.c

```diff
@@ -565,13 +565,6 @@ dnf_utils_add_remote (PkBackendJob *job,
 	if (!ret)
 		return FALSE;
 
-	/* update the AppStream copies in /var */
-	for (guint i = 0; i < repos->len; i++) {
-		DnfRepo *repo = g_ptr_array_index (repos, i);
-		if (!dnf_utils_refresh_repo_appstream (repo, error))
-			return FALSE;
-	}
-
```

Comment:
So this deletes the `dnf_utils_refresh_repo_appstream()` calls from add_remote(). The only other place where it's called from is `pk_backend_refresh_cache_thread` -> `pk_backend_refresh_repo` ->  `dnf_utils_refresh_repo_appstream`. Not sure if we are hitting that path all the time? I think gnome-software may explicitly call `pk_client_refresh_cache`, but not sure if all other frontends do.

Reply:
>  dnf_utils_refresh_repo_appstream

This is only "hit" when we download new metadata, which of course is fine to write a new appstream file.

> I think gnome-software may explicitly call pk_client_refresh_cache, but not sure if all other frontends do.

I guess maybe we could test if the file already exists? It does seem a bit of a layering violation in that case as libappstream-glib would need to leak out the appstream system path. I'm also not sure how much I care about other front-ends...